packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ButtonHTMLAttributes, Fragment, HTMLAttributes, memo, ReactNode, useMemo } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown, DropdownProps } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface ButtonGroupItem extends ButtonHTMLAttributes<HTMLButtonElement> {
+  content: string | ReactNode;
+}
+
+export interface ButtonGroupDropdownItem extends Omit<DropdownProps, 'toggle'> {
+  toggle: ButtonGroupItem;
+}
+
+export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement>, MarginProps {
+  items: Array<ButtonGroupItem | ButtonGroupDropdownItem>;
+}
+
+const isDropdownItem = (item: ButtonGroupItem | ButtonGroupDropdownItem): item is ButtonGroupDropdownItem =>
+  'items' in item && Array.isArray(item.items);
+
+const getButtonProps = (item: ButtonGroupItem | ButtonGroupDropdownItem) => (isDropdownItem(item) ? item.toggle : item);
+
+const ButtonItem: React.FC<ButtonGroupItem & { isFirst: boolean; isLast: boolean }> = memo(
+  ({ content, isFirst, isLast, ...restProps }) => {
+    if (typeof content === 'string') {
+      return (
+        <StyledButton {...restProps} variant="secondary" isFirst={isFirst} isLast={isLast}>
+          {content}
+        </StyledButton>
+      );
+    }
+
+    return <StyledButton {...restProps} variant="secondary" isFirst={isFirst} isLast={isLast} iconOnly={content} />;
```

Comment:
Can we have an icon and text as part of content in a button?

Reply:
I talk to Eugen Polev about this requirement... so, we can't use `icon` + `text` at the same time

Reply:
Also, we have a style restrictions... only `secondary` variant.

Reply:
I think we can allow icon and text option but in our guidelines we will recommend using text for all buttons in the group except for the ellipsis for more actions

Reply:
@eugene-polev so, should I add possibility to use `icon` + `text`? or should we restrict it?

Reply:
We already have it built into our `Button` component that if the `iconOnly` prop is present, the text content will not show. From my previous comment above, I think we should use the same pattern with `actions` in our `Modal` component to consume an array of `ButtonProps & { text: string; }`. That way the consumer can add the `iconOnly` prop if needed.

In combination with this component handling the overflow, we should enforce `text` to always be present so it can populate the dropdown, even if the button has the `iconOnly` prop.